app/config.py

```diff
@@ -0,0 +1,13 @@
+"""
+Load environment variables from the .env file.
+"""
+
+from pydantic_settings import BaseSettings, SettingsConfigDict
+
+
+class Settings(BaseSettings):
+    api_protocol: str = "http"
+    api_domain: str = "localhost"
+    api_port: int = 8000
```

Comment:
You could simply use [`AnyHttpUrl`](https://docs.pydantic.dev/2.2/api/networks/#pydantic.networks.AnyHttpUrl)